arduino/src/arduino.cpp

```diff
@@ -0,0 +1,285 @@
+    {
+        serial.closeDevice();
+        return Arduino::ERROR_ARDUINO::SUCCSESS;
+    }
+    serial.closeDevice();
+    return Arduino::ERROR_ARDUINO::SUCCSESS;
+}
+/**
+ * @brief check the connection for debugging
+ */
+void Arduino::checkConnectionToConsole()
+{
+    if (checkConnection() != Arduino::ERROR_ARDUINO::SUCCSESS)
+    {
+        std::cout << "there was a problem with the connection" << std::endl;
+    }
+    std::cout << "the connection is successful!-" << std::endl;
+}
+/**
+ * @brief add file to the files array
+ * @param file add a file to the array of files
+ */
+
+void Arduino::addFile(std::string file)
+{
+    this->files.push_back(file);
+}
+/*
+ * @brief will calculate the sum of a string
+ * I forgot that I found this function on the web so I made 
+ * a simple version of this using std::string
+ * this one will probably be used in the arduino
+ * @param message char array pointer 
+ * @return int the sum of the chars ascii values.
+ */
+int Arduino::checkSum(char *message)
+{
+    int sum = 0;
+    char *p = &message[0];
+    while (*p != NULLCHAR)
+    {
+        sum += *message;
+        ++p;
+    }
+    return sum;
+}
+/**
+ * @brief open the serial class with the members of this class
+ * @return Arduino::ERROR_ARDUINO will return the error code if there is one. 
+ */
+Arduino::ERROR_ARDUINO Arduino::openSerial()
+{
+    int err = this->serial.openDevice(this->portName.c_str(), this->buadRate);
+    if (err != SUCCESS)
+        return Arduino::ERROR_ARDUINO::ERROR_WITH_SERIALPORT;
+    this->isOpen = true;
+    return Arduino::ERROR_ARDUINO::SUCCSESS;
+}
+/*
+ * @brief send a serial commend to the arduino and confirm
+ *  that the message has been received 
+ * 
+ * @param message  the commend to be sent, for now 3 is the prefix 
+ * of the commends and the 2 digits behind it are the commend itself
+ * 
+ * @return the error code.
+ */
+Arduino::ERROR_ARDUINO Arduino::serialCommend(std::string message)
+{
+    if (!this->isOpen)
+        return Arduino::ERROR_ARDUINO::ERROR_WITH_SERIALPORT;
+    char inputBuff[CHECKBUFFSIZE] = {0};
+    this->serial.writeString(message.c_str());
+    // check sum should be added.
+    return Arduino::ERROR_ARDUINO::SUCCSESS;
+}
+/**
+ * @brief dumps data to file +time stamp
+ * @param data the data from the arduino 
+ * @param place from the file array
+ * @return Arduino::ERROR_ARDUINO will return the error code if there is one. 
+ */
+Arduino::ERROR_ARDUINO Arduino::writeFromBufferToFile(std::string data, int place)
+{
+    std::ofstream dataLog; // this part could be one line but I want it that way.
+    std::string fileName = "../dataFromArduino/sensor";
+    std::string fileNameEnd = ".log";
+    fileName.append(std::to_string(place));
+    fileName.append(fileNameEnd);
+    dataLog.open(fileName, std::ios_base::app);
+    if (!dataLog)
+    {
+        return Arduino::ERROR_ARDUINO::ERROR_WITH_FILES;
+    }
+    dataLog << std::time(0) << std::endl;
```

Comment:
I'm pretty sure it should be that
```suggestion
    dataLog << std::time(nullptr) << std::endl;
```

Reply:
fixed